src/background/Wallet/WalletRecord.ts

```diff
@@ -718,13 +718,21 @@ export class WalletRecordModel {
     return createChain(chain || NetworkId.Ethereum);
   }
 
-  static getPermissionsByChain(
+  static removeChainFromPermissions(
     record: WalletRecord,
     { chain }: { chain: Chain }
   ) {
-    return Object.entries(record.permissions)
-      .filter(([, permission]) => permission.chain === chain.toString())
-      .map(([origin, permission]) => ({ origin, permission }));
+    return produce(record, (draft) => {
+      draft.permissions = Object.fromEntries(
+        Object.entries(record.permissions).map(([origin, permission]) => {
+          if (permission.chain === chain.toString()) {
+            return [origin, { addresses: permission.addresses }];
+          } else {
+            return [origin, permission];
+          }
+        })
+      );
+    });
```

Comment:
The idea of `immer` is that you can mutate where it's convenient:
```suggestion
    return produce(record, (draft) => {
      for (const permission of Object.values(draft.permissions)) {
        if (permission.chain === chain.toString()) {
          delete permission.chain;
        }
      }
    });
```

Reply:
Nice, I need to get used to that, thanks!